### PR TITLE
Support extended selectors for sriov-device-plugin with helm deployment

### PR DESCRIPTION
**Pull request by almaslennikov**
Currently the SR-IOV Device Plugin resource definition can only select devices based on vendor ID. This creates problems in heterogeneous cluster deployments.
The proposed solution is to support all selectors from sriov-device-plugin.
The fix is intentionally aligned with the current implementation and not made more generic as that would require changing `values.yaml` and ensuring backwards compatibility with different versions of network-operator

### CODE REVIEW

**Comment by AbdYsn** [comment]
There was a power issue in the labs, retesting the CIs /retest-nic_operator_helm /retest-nic_operator_kind

**Review by e0ne — COMMENTED** [review]
LGTM. Let's wait until CI passes and we cut v1.1.0 release

File: deployment/network-operator/README.md [hunk 1]
```diff
@@ -393,6 +393,13 @@ to be associated with the resource. Refer to [SR-IOV Network Device Plugin Selec
 resources:
     - name: hostdev
       vendors: [15b3]
+    - name: infiniband_rdma
+      vendors: [15b3]
+      linkTypes: [infiniband]
```
Comment (adrianchiris) [inline deployment/network-operator/README.md:398]: lets keep this example with ethernet

File: deployment/network-operator/values.yaml [hunk 2]
```diff
@@ -158,6 +158,12 @@ sriovDevicePlugin:
   resources:
     - name: hostdev
       vendors: [15b3]
+    - name: infiniband_rdma
```
Comment (adrianchiris) [inline deployment/network-operator/values.yaml:161]: This affects the default configuration right ? i dont think we want that.

also these 3 resources now collide 

1 includes 2 and 2 
2 may include 3

**Comment by adrianchiris** [comment]
@almaslennikov also please fix commit message

File: deployment/network-operator/values.yaml (hunk 2 above)
Comment (almaslennikov) [reply deployment/network-operator/values.yaml]: Let's keep the default configuration as is and only change the examples in readme

File: deployment/network-operator/templates/mellanox.com_v1alpha1_nicclusterpolicy_cr.yaml [hunk 3]
```diff
@@ -84,6 +84,10 @@ spec:
             "resourceName": {{ $element.name | quote }},
             "selectors": {
               "vendors": {{ $element.vendors | default list | toJson }},
+              "devices": {{ $element.devices | default list | toJson }},
+              "drivers": {{ $element.drivers | default list | toJson }},
+              "ifNames": {{ $element.ifNames | default list | toJson }},
```
Comment (adrianchiris) [inline deployment/network-operator/templates/mellanox.com_v1alpha1_nicclusterpolicy_cr.yaml:89]: sriov network device plugin does not have this selector:  https://github.com/k8snetworkplumbingwg/sriov-network-device-plugin#extended-selectors-for-device-type-netdevice

Comment (adrianchiris) [reply deployment/network-operator/templates/mellanox.com_v1alpha1_nicclusterpolicy_cr.yaml:89]: https://github.com/k8snetworkplumbingwg/sriov-network-device-plugin#device-selectors

Comment (adrianchiris) [reply deployment/network-operator/templates/mellanox.com_v1alpha1_nicclusterpolicy_cr.yaml:89]: we should align to have the following selectors.

vendors, devices, drivers, pfNames, pciAddresses, rootDevices, linkTypes

**Pull request merged**